browsergym/core/src/browsergym/core/env.py

```diff
@@ -455,8 +471,9 @@ def _task_validate(self):
         prev_active_page = self.page
         prev_page_history = self.page_history.copy()
         # call validate
-        reward, done, user_message, info = self.task.validate(self.page, self.chat.messages)
-
+        reward, done, user_message, info = self.task.validate(
+            self.page, self.chat.messages, self.last_action
```

Comment:
There is no need for this. The idea of using Javascript to register actions executed in the browser is so that there is no need any more for the environment to give the executed `last_action` to the task. The BrowserGym last action could be bid-based or coord-based, which does not match the XPath-based validation of WebCanvas, as far as I understand. By using an event listener + playwright callback instead, the task can keep track of all the JS events and validate that they correspond to the task's objectives.
```suggestion
            self.page, self.chat.messages
```